### ISSUE TITLE
Reconsider checking `slog.With` and `slog.Logger.With`

### ISSUE THREAD

**Issue by AlekSi** [go-simpler/sloglint#49]
With `kv-only: false, attr-only: true` calls like `l.With("a", "b")` are reported as "key-value pairs should not be used". But there is no non-kv API for that operation.

**Comment by tmzane** [comment]
> But there is no non-kv API for that operation.

```go
slog.With(slog.String("a", "b")).Info("test")
```

@AlekSi the above works for me 🤔

**Comment by AlekSi** [comment]
Doh! You right, sorry for the noise
